### DISCUSSION TITLE
Is it possible to capture a part of a note and insert it below a certain heading?

### DISCUSSION THREAD

**Discussion by mpas** [zk-org/zk-nvim#83] [Help] [answered]
I am happily using zk for my notes. Lately I have a need where my todo list is also a zk note and was wondering if it is possible to create a part of a note and insert this below a certain heading?

So that I quickly can capture a note and insert this as a todo item below for example an # Inbox header?

**Comment by mickael-menu** [comment]
It's probably not possible at the moment.

So that I understand better, you want to generate a new note with `zk`, using a template, but insert it at the Vim cursor location instead of creating a new file in the notebook?

**Reply by mpas** [reply]
I have moved from Emacs/Org-Mode where I could quickly capture a TODO and this would be put below a certain header for later processing. Like an INBOX header for example. It would be great if I could capture a quick TODO with ZK and put this below a certain header comparable to how I used Org-Mode. 

So your assumption is correct that I would like to create a partial note and insert it in an already existing note. What I have now is a simple function that opens my ZK inbox note and puts the cursor at the last line. 

```
local function open_inbox()
    require("zk.commands").get("ZkNew")({ title = "inbox", dir = "" })
    vim.wait(100)
    vim.fn.feedkeys("G$a")
end
```

I can totally imagine that the scope what I am asking goes beyond the intention of ZK. :)

**Reply by mickael-menu** [reply]
Yes, I think it's a bit out of scope for `zk`. However, you could generate a new note to `/dev/stdout` without creating it on the file system with:

```
zk new --dry-run 2> /dev/null
```

(the generated new path is printed to `/std/err`, hence `2> /dev/null`).

With this, you could write a script to create a new note and insert it where you need.

The `--dry-run` flag is not available with the LSP `zk.new` command, but could be if needed.

**Comment by mickael-menu** [comment, marked as answer]
I believe that this new PR should allow you to do that: https://github.com/mickael-menu/zk-nvim/pull/92

```vim
:ZkNew { inline = true, template = "part.md" }
```

**Reply by mpas** [reply]
Thanks a lot for the update!!
